lib/install.go

```diff
@@ -50,10 +48,10 @@ func GetInstallLocation(installPath string) string {
 }
 
 // install : install the provided version in the argument
-func install(product Product, tfversion string, binPath string, installPath string, mirrorURL string) error {
+func install(product Product, tfversion, binPath, installPath, mirrorURL, goarch string) error {
```

Comment:
Maybe I've missed something (I don't follow golang), why has `string` type been removed?

Reply:
`string` type is not removed. it's "optimized". in Go when you provide list of parameters for a function and if they are of the same type, you may specify type at the end of the list. so in this example the `product` is of type `Product` and the rest are of the type `string` since `string` is specified at the end of the string parameters list.

Reply:
The guy explained it in details here: https://stackoverflow.com/a/76593363/5093149

Reply:
Please let me know if you're ok with this feature.

Reply:
Yep, if it works, seems fine